<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>961</width>
    <height>720</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <layout class="QHBoxLayout" name="hlMain">
      <item>
       <layout class="QVBoxLayout" name="vbControls">
        <item>
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Controls</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="Sliders">
          <item>
           <layout class="QVBoxLayout" name="Slider1">
            <item>
             <widget class="QLabel" name="labelFOV">
              <property name="text">
               <string>Field of View</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSpinBox" name="sbFOV">
              <property name="minimum">
               <number>45</number>
              </property>
              <property name="maximum">
               <number>180</number>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSlider" name="vsFOV">
              <property name="minimum">
               <number>45</number>
              </property>
              <property name="maximum">
               <number>180</number>
              </property>
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
             </widget>
            </item>
           </layout>
          </item>
          <item>
           <layout class="QVBoxLayout" name="Slider2">
            <item>
             <widget class="QLabel" name="lableAngle">
              <property name="text">
               <string>Angle</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSpinBox" name="sbAngle">
              <property name="maximum">
               <number>360</number>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSlider" name="vsAngle">
              <property name="maximum">
               <number>360</number>
              </property>
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
             </widget>
            </item>
           </layout>
          </item>
         </layout>
        </item>
        <item>
         <layout class="QVBoxLayout" name="butto_layout">
          <item>
           <widget class="QLabel" name="pro_label">
            <property name="text">
             <string>Projection Mode</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="rbPerspective">
            <property name="text">
             <string>Perspective</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QRadioButton" name="rbOrthogonal">
            <property name="text">
             <string>Orthogonal</string>
            </property>
           </widget>
          </item>
          <item>
           <layout class="QVBoxLayout" name="rotbox">
            <item>
             <layout class="QFormLayout" name="rotation_layout">
              <item row="0" column="0">
               <widget class="QLabel" name="Rotation_Label">
                <property name="text">
                 <string>Rotations</string>
                </property>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="QLabel" name="labelA">
                <property name="text">
                 <string>Ring A</string>
                </property>
               </widget>
              </item>
              <item row="2" column="0">
               <widget class="QLabel" name="labelB">
                <property name="text">
                 <string>Ring B</string>
                </property>
               </widget>
              </item>
              <item row="2" column="1">
               <widget class="QSlider" name="hsRotationB">
                <property name="maximum">
                 <number>180</number>
                </property>
                <property name="singleStep">
                 <number>1</number>
                </property>
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
               </widget>
              </item>
              <item row="3" column="0">
               <widget class="QLabel" name="labelC">
                <property name="text">
                 <string>Ring C</string>
                </property>
               </widget>
              </item>
              <item row="3" column="1">
               <widget class="QSlider" name="hsRotationC">
                <property name="maximum">
                 <number>180</number>
                </property>
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
               </widget>
              </item>
              <item row="1" column="1">
               <widget class="QSlider" name="hsRotationA">
                <property name="maximum">
                 <number>180</number>
                </property>
                <property name="orientation">
                 <enum>Qt::Horizontal</enum>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <widget class="QCheckBox" name="checkBox">
              <property name="text">
               <string>Animate</string>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QCheckBox" name="checkBox_2">
              <property name="text">
               <string>Camera</string>
              </property>
             </widget>
            </item>
            <item>
             <layout class="QVBoxLayout" name="verticalLayout">
              <item>
               <layout class="QFormLayout" name="Near_Layout">
                <item row="0" column="0">
                 <widget class="QLabel" name="clipping_label">
                  <property name="text">
                   <string>Clipping Plane</string>
                  </property>
                 </widget>
                </item>
                <item row="1" column="0">
                 <widget class="QLabel" name="near_label">
                  <property name="text">
                   <string>Near</string>
                  </property>
                 </widget>
                </item>
                <item row="1" column="1">
                 <widget class="QDoubleSpinBox" name="sbNear">
                  <property name="value">
                   <double>0.100000000000000</double>
                  </property>
                 </widget>
                </item>
                <item row="2" column="0">
                 <widget class="QLabel" name="far_label">
                  <property name="text">
                   <string>Far</string>
                  </property>
                 </widget>
                </item>
                <item row="2" column="1">
                 <widget class="QDoubleSpinBox" name="sbFar">
                  <property name="value">
                   <double>99.900000000000006</double>
                  </property>
                 </widget>
                </item>
               </layout>
              </item>
             </layout>
            </item>
            <item>
             <layout class="QFormLayout" name="_2">
              <item row="0" column="1">
               <widget class="QPushButton" name="bReset">
                <property name="text">
                 <string>Reset</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item>
       <widget class="MyGLWidget" name="openGLWidget">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="minimumSize">
         <size>
          <width>720</width>
          <height>480</height>
         </size>
        </property>
       </widget>
      </item>
     </layout>
    </item>
   </layout>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>MyGLWidget</class>
   <extends>QOpenGLWidget</extends>
   <header>myglwidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>vsFOV</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>sbFOV</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>24</x>
     <y>189</y>
    </hint>
    <hint type="destinationlabel">
     <x>60</x>
     <y>65</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>vsAngle</sender>
   <signal>sliderMoved(int)</signal>
   <receiver>sbAngle</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>136</x>
     <y>153</y>
    </hint>
    <hint type="destinationlabel">
     <x>170</x>
     <y>66</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sbFOV</sender>
   <signal>valueChanged(int)</signal>
   <receiver>vsFOV</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>85</x>
     <y>67</y>
    </hint>
    <hint type="destinationlabel">
     <x>33</x>
     <y>238</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sbAngle</sender>
   <signal>valueChanged(int)</signal>
   <receiver>vsAngle</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>195</x>
     <y>68</y>
    </hint>
    <hint type="destinationlabel">
     <x>140</x>
     <y>225</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
